From 0059c7ed6c11254f214a07357bbf87296cde446f Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Wed, 7 Nov 2018 22:19:07 +0800
Subject: [PATCH] MLK-20205-1: dmaengine: fsl-edma-v3: fix NULL pointer
 dereference

Fix 'null pointer dereferences issue' reported by coverity(CID-1477441).

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Reviewed-by: Anson Huang <Anson.Huang@nxp.com>
(cherry picked from commit 5343c0018af0af2eb3bb90c5a75e765b851a2c12)
---
 drivers/dma/fsl-edma-v3.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/dma/fsl-edma-v3.c
+++ b/drivers/dma/fsl-edma-v3.c
@@ -419,7 +419,7 @@ static enum dma_status fsl_edma3_tx_stat
 	if (fsl_chan->edesc && cookie == fsl_chan->edesc->vdesc.tx.cookie)
 		txstate->residue = fsl_edma3_desc_residue(fsl_chan, vdesc,
 								true);
-	else if (vdesc)
+	else if (fsl_chan->edesc && vdesc)
 		txstate->residue = fsl_edma3_desc_residue(fsl_chan, vdesc,
 								false);
 	else
